File: src/contests/cg/puzzle/haskell/distributing-candy.hs
-- 🎮 CodinGame Puzzle - distributing-candy
-- https://www.codingame.com/training/easy/distributing-candy

import Data.List (sort)

main :: IO ()
main = do
  l1 <- getLine
  let [_, m] = map read (words l1) :: [Int]
  l2 <- getLine
  let sorted = sort (map read (words l2)) :: [Int]
  print (minimum (zipWith (-) (drop (m - 1) sorted) sorted))

File: src/contests/cg/puzzle/lua/distributing-candy.lua
-- 🎮 CodinGame Puzzle - distributing-candy
-- https://www.codingame.com/training/easy/distributing-candy

local sn, sm = io.read():match("(%S+)%s+(%S+)")
local n, m = tonumber(sn), tonumber(sm)
local candies = {}
for num in io.read():gmatch("%S+") do
  candies[#candies + 1] = tonumber(num)
end
table.sort(candies)
local best = math.huge
for i = 1, n - m + 1 do
  local diff = candies[i + m - 1] - candies[i]
  if diff < best then
    best = diff
  end
end
print(best)
